transports/websocket/src/framed.rs

```diff
@@ -53,7 +58,11 @@ pub struct WsConfig<T> {
     tls_config: tls::Config,
     max_redirects: u8,
     use_deflate: bool,
-    listener_protos: HashMap<ListenerId, Protocol<'static>>,
+    /// Ws / Wss protocols extension for inner listening addresses, mapped to
+    /// the port of the listening address.
+    listener_protos: HashMap<u16, Protocol<'static>>,
```

Comment:
This could collide on IPv6 and IPv4, right? E.g. say I am listening via `wss` on `/ip6/xxx/tcp/42` and `ws` on `/ip4/xxx/tcp/42`.

Reply:
Good catch!